
# HG changeset patch
# User serge-sans-paille <sguelton@mozilla.com>
# Date 1678865227 0
# Node ID cbc55e58a4b14ab484486c72a47f595fa223336b
# Parent  7d86c3224520ce8b5c53ebd7122671f0f69fb796
Bug 1821363 - Use fully specified xsimd::batch type r=padenot, a=dsmith

That way we cannot have one type using the default architecture that
interacts with the parametric type required by the user.

Differential Revision: https://phabricator.services.mozilla.com/D172126

diff --git a/dom/media/webaudio/AudioNodeEngineGeneric.h b/dom/media/webaudio/AudioNodeEngineGeneric.h
--- a/dom/media/webaudio/AudioNodeEngineGeneric.h
+++ b/dom/media/webaudio/AudioNodeEngineGeneric.h
@@ -200,20 +200,20 @@ struct Engine {
     MOZ_ASSERT(is_aligned<Arch>(aInput), "aInput is aligned");
     MOZ_ASSERT(is_aligned<Arch>(aOutput), "aOutput is aligned");
     MOZ_ASSERT(is_aligned<Arch>(aScale), "aScale is aligned");
     MOZ_ASSERT((aSize % xsimd::batch<float, Arch>::size == 0),
                "requires tail processing");
 
     MOZ_UNROLL(2)
     for (unsigned i = 0; i < aSize * 2;
-         i += 2 * xsimd::batch<std::complex<float>>::size) {
-      auto in1 = xsimd::batch<std::complex<float>>::load_aligned(
+         i += 2 * xsimd::batch<std::complex<float>, Arch>::size) {
+      auto in1 = xsimd::batch<std::complex<float>, Arch>::load_aligned(
           reinterpret_cast<const std::complex<float>*>(&aInput[i]));
-      auto in2 = xsimd::batch<std::complex<float>>::load_aligned(
+      auto in2 = xsimd::batch<std::complex<float>, Arch>::load_aligned(
           reinterpret_cast<const std::complex<float>*>(&aScale[i]));
       auto out = in1 * in2;
       out.store_aligned(reinterpret_cast<std::complex<float>*>(&aOutput[i]));
     }
   };
 
   static float AudioBufferSumOfSquares(const float* aInput, uint32_t aLength) {
     float sum = 0.f;

